elastalert/util.py

```diff
@@ -363,6 +363,11 @@ def build_es_conn_config(conf):
     parsed_conf['headers'] = None
     parsed_conf['es_host'] = os.environ.get('ES_HOST', conf['es_host'])
     parsed_conf['es_port'] = int(os.environ.get('ES_PORT', conf['es_port']))
+
+    es_hosts = os.environ.get('ES_HOSTS')
+    es_hosts = parse_hosts(es_hosts, parsed_conf.get('es_port')) if es_hosts else conf.get('es_hosts')
```

Comment:
Could we only parse and assign `es_hosts` if there is an environment variable?

Reply:
Pretty sure this will always stop the yml value.  Similar cases pass the default through

```
os.environ.get('ES_HOST', conf['es_host'])
```

Reply:
Instead of passing `parsed_conf.get('es_port')`, should we pass the fully resolved value from a couple lines above

```
parsed_conf['es_port']
```

Reply:
> We should only parse and assign `es_hosts` if there is an environment variable?

No, it only parses the value if it came from an environment variable. If the environment variable was not set then the `else conf.get('es_hosts') applies. I added unit tests yesterday showing this behavior.

Reply:
> Instead of passing `parsed_conf.get('es_port')`, should we pass the fully resolved value from a couple lines above
> 
> ```
> parsed_conf['es_port']
> ```
> 
> That way things work as expected if the port is set in yml but the hosts set via environment variable.

Your suggested `parsed_conf['es_port']` resolves to the same value as the implemented `parsed_conf.get('es_port')`, except that the implemented statement is safer since it won't throw an exception if the key doesn't exist in the map.

Reply:
Sorry completely missed the `if es_hosts else conf.get('es_hosts')` at the end of the line.  Consider simplifying.

Reply:
```
es_hosts = os.environ.get('ES_HOSTS')
if es_hosts:
    parsed_conf['es_hosts'] = parse_hosts(es_hosts, parsed_conf.get('es_port'))
```

This avoids a bonus read and write to `parsed_conf['es_hosts']` when configured by yml.